addons/microsoft_calendar/models/microsoft_sync.py

```diff
@@ -377,7 +378,12 @@ def _sync_microsoft2odoo(self, microsoft_events: MicrosoftEvent):
                 odoo_event_updated_time = pytz.utc.localize(odoo_event.write_date)
                 ms_event_updated_time = parse(mevent.lastModifiedDateTime)
 
-                if ms_event_updated_time >= odoo_event_updated_time:
+                # If the update comes from an old event, check if time diff between updates is reasonable.
+                # All 'calendar.recurrence' records are skipped since they don't have the 'stop' variable defined.
+                old_event_update_condition = True
+                if odoo_event._name == 'calendar.event':
+                    old_event_update_condition = odoo_event._check_old_event_update_required(odoo_event_updated_time, ms_event_updated_time)
```

Comment:
You will check the ICP value for all events. Thi is bad, you should get that value before entering the loop. lower_bound_day_range should be taken before the loop and passed as a parameter of the check function.